bin/docker/kafka_bridge_config_generator.sh

```diff
@@ -31,26 +31,47 @@ EOF
     fi
 fi
 
-if [ -n "$KAFKA_BRIDGE_SASL_USERNAME" ] && [ -n "$KAFKA_BRIDGE_SASL_PASSWORD_FILE" ]; then
+if [ -n "$KAFKA_BRIDGE_SASL_MECHANISM" ]; then
     if [ "$SECURITY_PROTOCOL" = "SSL" ]; then
         SECURITY_PROTOCOL="SASL_SSL"
     else
         SECURITY_PROTOCOL="SASL_PLAINTEXT"
     fi
 
-    PASSWORD=$(cat /opt/strimzi/bridge-password/$KAFKA_BRIDGE_SASL_PASSWORD_FILE)
-
     if [ "x$KAFKA_BRIDGE_SASL_MECHANISM" = "xplain" ]; then
+        PASSWORD=$(cat /opt/strimzi/bridge-password/$KAFKA_BRIDGE_SASL_PASSWORD_FILE)
         SASL_MECHANISM="PLAIN"
-        JAAS_SECURITY_MODULE="plain.PlainLoginModule"
+        JAAS_CONFIG="org.apache.kafka.common.security.plain.PlainLoginModule required username=\"${KAFKA_BRIDGE_SASL_USERNAME}\" password=\"${PASSWORD}\";"
     elif [ "x$KAFKA_BRIDGE_SASL_MECHANISM" = "xscram-sha-512" ]; then
+        PASSWORD=$(cat /opt/strimzi/bridge-password/$KAFKA_BRIDGE_SASL_PASSWORD_FILE)
         SASL_MECHANISM="SCRAM-SHA-512"
-        JAAS_SECURITY_MODULE="scram.ScramLoginModule"
+        JAAS_CONFIG="org.apache.kafka.common.security.scram.ScramLoginModule required username=\"${KAFKA_BRIDGE_SASL_USERNAME}\" password=\"${PASSWORD}\";"
+    elif [ "x$KAFKA_BRIDGE_SASL_MECHANISM" = "xoauth" ]; then
+        if [ ! -z "$KAFKA_BRIDGE_OAUTH_ACCESS_TOKEN" ]; then
+            OAUTH_ACCESS_TOKEN="oauth.access.token=\"$KAFKA_BRIDGE_OAUTH_ACCESS_TOKEN\""
+        fi
+
+        if [ ! -z "$KAFKA_BRIDGE_OAUTH_REFRESH_TOKEN" ]; then
+            OAUTH_REFRESH_TOKEN="oauth.refresh.token=\"$KAFKA_BRIDGE_OAUTH_REFRESH_TOKEN\""
+        fi
+
+        if [ ! -z "$KAFKA_BRIDGE_OAUTH_CLIENT_SECRET" ]; then
+            OAUTH_CLIENT_SECRET="oauth.client.secret=\"$KAFKA_BRIDGE_OAUTH_CLIENT_SECRET\""
+        fi
+
+        if [ -f "/tmp/strimzi/oauth.truststore.p12" ]; then
+            OAUTH_TRUSTSTORE="oauth.ssl.truststore.location=\"/tmp/strimzi/oauth.truststore.p12\" oauth.ssl.truststore.password=\"${CERTS_STORE_PASSWORD}\" oauth.ssl.truststore.type=\"PKCS12\""
+        fi
+
+        SASL_MECHANISM="OAUTHBEARER"
```

Comment:
can you move this line up having the same pattern as the other SASL mechanism blocks. Just setting the SASL_MECHANISM as the first thing.